File: app/src/main/java/ru/evgeniy/nytimes/Activity/IntroActivity.java
package ru.evgeniy.nytimes.Activity;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;

import java.util.concurrent.TimeUnit;

import io.reactivex.Completable;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;
import me.relex.circleindicator.CircleIndicator;
import ru.evgeniy.nytimes.App;
import ru.evgeniy.nytimes.fragments.IntroFragment;
import ru.evgeniy.nytimes.R;
import ru.evgeniy.nytimes.data.SharedPref;
import ru.evgeniy.nytimes.news.MainActivity;

public class IntroActivity extends AppCompatActivity {
    private static final int DELAY_TIME = 60;
    private static final int PAGE_COUNT = 3;
    private CompositeDisposable compositeDisposable = new CompositeDisposable();

    SharedPref Storage = App.getSharedPref();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_intro);
        CircleIndicator circleIndicator = findViewById(R.id.indicator);

        ViewPager viewpager = findViewById(R.id.viewPager);
        viewpager.setAdapter(new FragmentPagerAdapter(getSupportFragmentManager()) {
            @Override
            public Fragment getItem(int i) {

                Fragment fragment;
                switch (i){
                    case 0: fragment = new IntroFragment().newInstance(1);break;
                    case 1: fragment = new IntroFragment().newInstance(3);break;
                    default:  fragment = new IntroFragment().newInstance(2);break;
                }
                return fragment;
            }

            @Override
            public int getCount() {
                return PAGE_COUNT;
            }
        });
        circleIndicator.setViewPager(viewpager);

        if (Storage.needToShowIntro()) {
            Disposable disposable = Completable.complete()
                    .delay(DELAY_TIME, TimeUnit.SECONDS)
                    .subscribe(this::startMainActivity);
            compositeDisposable.add(disposable);
        } else {
            startMainActivity();
        }
    }

    @Override
    protected void onStop() {
        super.onStop();
        compositeDisposable.dispose();
    }

    private void startMainActivity() {
        startActivity(MainActivity.Companion.create(this));
        finish();
    }
}

File: app/src/main/java/ru/evgeniy/nytimes/activity/IntroActivity.kt
package ru.evgeniy.nytimes.activity

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentPagerAdapter
import android.support.v4.view.ViewPager
import android.support.v7.app.AppCompatActivity

import java.util.concurrent.TimeUnit

import io.reactivex.Completable
import io.reactivex.disposables.CompositeDisposable
import me.relex.circleindicator.CircleIndicator
import ru.evgeniy.nytimes.App
import ru.evgeniy.nytimes.fragments.IntroFragment
import ru.evgeniy.nytimes.R
import ru.evgeniy.nytimes.news.MainActivity

class IntroActivity : AppCompatActivity() {

    companion object {
        private const val DELAY_TIME = 3
        private const val PAGE_COUNT = 3
    }

    private val compositeDisposable = CompositeDisposable()
    private var storage = App.sharedPref

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_intro)
        val circleIndicator = findViewById<CircleIndicator>(R.id.indicator)

        val viewpager = findViewById<ViewPager>(R.id.viewPager)
        viewpager.adapter = object : FragmentPagerAdapter(supportFragmentManager) {
            override fun getItem(i: Int): Fragment {

                return when (i) {
                    0 -> IntroFragment().newInstance(R.drawable.device)
                    1 -> IntroFragment().newInstance(R.drawable.device_read)
                    else -> IntroFragment().newInstance(R.drawable.device_edit)
                }
            }

            override fun getCount(): Int {
                return PAGE_COUNT
            }
        }
        circleIndicator.setViewPager(viewpager)

        if (storage.needToShowIntro()) {
            val disposable = Completable.complete()
                    .delay(DELAY_TIME.toLong(), TimeUnit.SECONDS)
                    .subscribe { this.startMainActivity() }
            compositeDisposable.add(disposable)
        } else {
            startMainActivity()
        }
    }

    override fun onStop() {
        super.onStop()
        compositeDisposable.clear()
    }

    private fun startMainActivity() {
        startActivity(MainActivity.create(this))
        finish()
    }
}